pymeasure/instruments/anaheimautomation/dpseriesmotorcontroller.py

```diff
@@ -0,0 +1,357 @@
+        get_process=lambda err: DPSeriesErrors(int(err)),
+    )
+
+    def check_errors(self):
+        """ Method to read the error codes register and log when an error is detected.
+        
+        :return error_code: one byte with the error codes register contents
+        """
+        current_errors = self.error_reg
+        if current_errors != 0:
+            logging.error("DP-Series motor controller error detected: %s" % current_errors)
+        return current_errors
+    
+    def __init__(self, resourceName, address=0, encoder_enabled=False, **kwargs):
+        """
+        Initialize communication with the motor controller with the address given by the 'address' parameter.
+        In addition to the keyword arguments that can be set for the Instrument base class, this class has the following
+        kwargs:
+
+        :param address: (int) Address that the motor controller uses for serial communiation.
+        :param encoder_enabled: (bool) Flag to indicate if the driver should use an encoder input to set its position
+                                 property.
+        """
+        self._address = address
+        self._encoder_enabled = encoder_enabled 
+
+        super().__init__(
+            resourceName,
+            "Anaheim Automation Stepper Motor Controller",
+            includeSCPI=False,
+            write_termination="\r",
+            read_termination="\r",
+            timeout=2000, 
+            **kwargs
+        )
+        
+        if isinstance(self.adapter, VISAAdapter):
+            self.adapter.connection.baud_rate = 38400
+    
+    @property
+    def encoder_enabled(self):
+        """ A boolean property to represent whether an external encoder is connected and should be used to set the
+            step_position property.
+        """
+        return self._encoder_enabled
+
+    @encoder_enabled.setter
+    def encoder_enabled(self, en):
+        self._encoder_enabled = bool(en)
+    
+    @property
+    def step_position(self):
+        """ Integer property representing the value of the motor position measured in steps counted by the motor
+            controller or, if encoder_enabled is set, the steps counted by an externally connected encoder. Note that in
+            the DP series motor controller instrument manuals, this property would be referred to as the 'absolute
+            position' while this driver implements a conversion between steps and absolute units for the `absolute
+            position` property. This property can be set.
+        """
+        if self._encoder_enabled:
+            pos = self.ask("VEP")
+        else:
+            pos = self.ask("VZ")    
+        return int(pos)
+    
+    @step_position.setter
+    def step_position(self, pos):
+        step_pos = int(pos)
+        if -8388607 < step_pos < 8388607:
+            self.write("P%i" % step_pos)
+            self.write("G")
+        else:
+            raise ValueError("Provided step position out of valid range!")
```

Comment:
you can use the built-in `validators.strict_range` here:
```
step_pos = strict_range(pos, (-8388607, 8388607))
self.write("P%i" % step_pos)
self.write("G")
```
(the int() cast is also done by the string formatter %i)